common/dynamicconfig/constants.go

```diff
@@ -1451,6 +1451,15 @@ limit is in addition to queuePendingTaskCriticalCount which controls when to unl
 prevent loading new tasks. Ideally this max count limit should not be hit and task unloading should happen once critical
 count is exceeded. But since queue action is async, we need this hard limit.
 NOTE: The outbound queue has a separate configuration: outboundQueuePendingTaskMaxCount.
+`,
+	)
+	QueuePredicateMaxDepth = NewGlobalIntSetting(
```

Comment:
would it make sense to do a map setting from queue type to max depth?

Reply:
Yes but I'd probably want to change this consistently for all of the queue settings.
And outbound queue is a bit special here so I'm okay giving it special treatment.

Reply:
I guess eventually we will remove the special treatment for outbound queue and use this one for all queues?

Reply:
We should discuss